Rework backups (#57)
DSD-DBS/capella-dockerimages: README.md

    readonly
```

### 8. Docker image `t4c/client/backup`

> The T4C client backup image imports a model from a TeamForCapella server and exports it to a Git repository. It can be used as a backup solution, e.g. as cronjob every night.

Reviewer: ```suggestion
The T4C client backup image imports a model from a TeamForCapella server and exports it to a Git repository. It can be used as a backup solution, for example, from a scheduled job.
```

To build the image, please run:

```zsh

> docker build -t $BASE/t4c/client/backup \
>     --build-arg BASE_IMAGE=$BASE/t4c/client/base \
>     backup

Reviewer: ```suggestion
docker build -t t4c/client/backup \
    --build-arg BASE_IMAGE=t4c/client/base \
    backup
```

Author: Good catch! Thanks :)

- `$GIT_USERNAME` with the git username if the repository is access protected. Leave empty, when no authentication is required.
- `$GIT_PASSWORD` with the git password if the repository is access protected. Leave empty, when no authentication is required. The password gets cleaned after cloning and is not accessible in the RDP connection.

### Backup image

> Please run the following command:

Reviewer: ```suggestion
Please run the following command to backup from T4C to Git:
```

It's more clear if the purpose of the command is stated as well, since people do not tend to read top-to-bottom.

```zsh
docker run -d \
  -e GIT_REPO_URL=https://github.com/example/example.git \
  -e GIT_REPO_BRANCH=main \
  -e GIT_USERNAME=user \
  -e GIT_PASSWORD=password \
  -e T4C_REPO_HOST=localhost \
  -e T4C_REPO_PORT=2036 \
  -e T4C_CDO_PORT=12036 \
  -e T4C_REPO_NAME=repoCapella \
  -e T4C_PROJECT_NAME=test \
  -e T4C_USERNAME=user \
  -e T4C_PASSWORD=password \
  -e LOG_LEVEL=DEBUG="DEBUG" \
  -e INCLUDE_COMMIT_HISTORY=false \
  t4c/client/backup
```

> Please set the following values for the corresponding keys:

Reviewer: ```suggestion
Set the following values for the corresponding keys:
```

Author: I just wanted to be polite :)

> - `GIT_REPO_URL` with the URL to the target Git repository (where the model is pushed to). All URI-formats supported by the `git clone` command will work. You can provide HTTP credentials via the `GIT_USERNAME` and `GIT_PASSWORD` variables (see below).

Author: Done.

> - `GIT_REPO_BRANCH` with the branch of the Git repository. We push to the corresponding branch.

Author: Because it's clear from the context?

> - `GIT_REPO_BRANCH` with the branch of the Git repository. We push to the corresponding branch.

Author: Will mark it as resolved for now, can be handled as separate PR if important.

> - `GIT_REPO_URL` with the URL to the target Git repository (where the model is pushed to). All URI-formats supported by the `git clone` command will work. You can provide HTTP credentials via the `GIT_USERNAME` and `GIT_PASSWORD` variables (see below).
> - `GIT_REPO_BRANCH` with the branch of the Git repository. We push to the corresponding branch.
> - `GIT_USERNAME` with the Git username if the repository is access protected.
> - `GIT_PASSWORD` with the Git password that is used during cloning from and pushing to the Git repository.
> - `T4C_REPO_HOST` with the hostname to the T4C server (the same value that you enter in Capella to connect to a remote repository)
> - `T4C_REPO_PORT` with the port to the T4C server (the same value that you enter in Capella to connect to a remote repository). Defaults to 2036.
> - `T4C_CDO_PORT` with the CDO port to the T4C server. Defaults to 12036.
> - `T4C_REPO_NAME` with the T4C repository name (the same value that you enter in Capella to connect to a remote repository)
> - `T4C_PROJECT_NAME` with the name of the Capella project. It's displayed in the Capella project explorer and in the last step when connecting to a remote repository.
> - `T4C_USERNAME` with the T4C username that is used during the import. The user needs to have access to the repository.
> - `T4C_PASSWORD` with the T4C password that is used during the import.
> - `LOG_LEVEL` to your preferred logging level (all Python logging levels are supported).
> - `INCLUDE_COMMIT_HISTORY` with `true` or `false` to define if the T4C commit history should be exported. Important: Exporting the commit history can take a few hours for large models.

Reviewer: ```suggestion
- `GIT_REPO_URL`: URL to the target Git repository where the model will be pushed to. All URI-formats supported by the `git clone` command will work. You can provide HTTP credentials via the `GIT_USERNAME` and `GIT_PASSWORD` variables (see below).
- `GIT_REPO_BRANCH`: branch of the Git repository.
- `GIT_USERNAME`: Git username if the repository is access protected.
- `GIT_PASSWORD`: Git password that is used during cloning from and pushing to the Git repository.
- `T4C_REPO_HOST`: hostname to the T4C server. The same value that you enter in Capella to connect to a remote repository.
- `T4C_REPO_PORT`: port to the T4C server. The same value that you enter in Capella to connect to a remote repository. Defaults to 2036.
- `T4C_CDO_PORT`: CDO port to the T4C server. Defaults to 12036.
- `T4C_REPO_NAME`: T4C repository name. The same value that you enter in Capella to connect to a remote repository.
- `T4C_PROJECT_NAME`: name of the Capella project. It's displayed in the Capella project explorer and in the last step when connecting to a remote repository.
- `T4C_USERNAME`: T4C username that is used during the import. The user needs to have access to the repository.
- `T4C_PASSWORD`: T4C password that is used during the import.
- `LOG_LEVEL`: your preferred logging level (all Python logging levels are supported).
- `INCLUDE_COMMIT_HISTORY`: `true` or `false` to define if the T4C commit history should be exported. Important: Exporting the commit history can take a few hours for large models.
```

Author: Done.
